portfolio/src/main/java/com/google/sps/servlets/GameServlet.java

```diff
@@ -0,0 +1,172 @@
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+
+import java.awt.image.BufferedImage;
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.Graphics;
+import javax.imageio.ImageIO;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.util.Base64;
+import java.util.HashMap;
+
+@WebServlet("/game")
+public class GameServlet extends HttpServlet {
+    private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Scores of games currently in progress.
+    private HashMap<Long, Integer> curScores = new HashMap<>();
+
+    // Next answers for games currently in progress.
+    private HashMap<Long, Integer> ans = new HashMap<>();
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        // Load request params into strings.
+        String startTimeStr = request.getParameter("start");
+        String userAnsStr = request.getParameter("ans");
+        String name = request.getParameter("name");
+
+        // A post request must have a game start time and either an answer or username (for the highscore list).
+        if (startTimeStr == null || (userAnsStr == null && name == null)) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return;
+        }
+
+        long startTime = Long.parseLong(startTimeStr);
```

Comment:
What happens if the input start time isn't a long?

Reply:
Added a try catch and "Invalid start time" alert on client side.